library/src/main/java/com/novoda/storagepathfinder/ExternalFileDirectoryInspector.java

```diff
@@ -88,7 +88,11 @@ private String getDirectoryPathAboveTheAndroidFolderFrom(File file) {
         return path == null ? "" : path;
     }
 
-    private boolean isValidPathForSecondaryStorage(String absolutePath) {
-        return !absolutePath.isEmpty() && !absolutePath.equals(primaryStoragePath);
+    private boolean isValidPathForSecondaryStorage(String absolutePath, Filter base) {
+        String pathToCompare = absolutePath;
+        if (base == Filter.APPLICATION) {
+            pathToCompare = getDirectoryPathAboveTheAndroidFolderFrom(new File(absolutePath));
```

Comment:
Would it be better to pass just the `absolutePath`, it feels unnecessary to create a `File` and then immediately get the `String`.

Reply:
I am going to refactor this whole class, should I bother with this change in the PR now?

Reply:
Nah, do it in another PR 😄